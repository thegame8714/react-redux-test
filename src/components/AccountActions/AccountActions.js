import React from 'react'

export default class AccountActions extends React.Component {

    static propTypes = {
        amount: React.PropTypes.string,
        maxAmount: React.PropTypes.string,
        topUpAccount: React.PropTypes.func,
        withdrawAccount: React.PropTypes.func
    }

    topUpAccount() {
        this.props.topUpAccount()
    }

    withdrawAccount() {
        this.props.withdrawAccount()
    }


    render() {
        const {
            amount,
            maxAmount
        } = this.props

        return (
            <div className="account-actions">
                <button className="topUp button" onClick={this.topUpAccount.bind(this)} disabled={parseFloat(amount) >=  parseFloat(maxAmount) ? "disabled" : ""}>TOP UP $1000 ON ACCOUNT</button>
                <button className="withdraw button" onClick={this.withdrawAccount.bind(this)} disabled={parseFloat(amount) === 0 ? "disabled" : ""}>WITHDRAW $1000 FROM ACCOUNT</button>
            </div>
        )
            

    }

}