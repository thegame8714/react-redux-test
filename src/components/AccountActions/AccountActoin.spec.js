import React from 'react'
import AccountActions from './AccountActions'
import { shallow } from 'enzyme'
import chai from 'chai'
import sinon from 'sinon'
import renderer from 'react-test-renderer'

describe('AccountActions', () => {
    it('should call the topUp function when click on the top up button', () => {
        const topUpAccount = sinon.spy()
        const accountActionsComponent = shallow(<AccountActions amount={"1000"}
                                                                maxAmount={"11000"} 
                                                                topUpAccount={topUpAccount} 
                                                                withdrawAccount={sinon.spy()} />)

        accountActionsComponent.find('.topUp').simulate('click')
        chai.expect(topUpAccount).to.be.called
    })

    it('should call the withdraw function when click on the top up button', () => {
        const withdrawAccount = sinon.spy()
        const accountActionsComponent = shallow(<AccountActions amount={"1000"}
                                                                maxAmount={"11000"} 
                                                                topUpAccount={sinon.spy()} 
                                                                withdrawAccount={withdrawAccount} />)

        accountActionsComponent.find('.withdraw').simulate('click')
        chai.expect(withdrawAccount).to.be.called
    })

     it('should render the page', () => {
        const accountActionsComponent = renderer.create(<AccountActions amount={"1000"}
                                                                maxAmount={"11000"} 
                                                                topUpAccount={sinon.spy()} 
                                                                withdrawAccount={sinon.spy()} />)
        
        expect(accountActionsComponent.toJSON()).toMatchSnapshot()
    })
})