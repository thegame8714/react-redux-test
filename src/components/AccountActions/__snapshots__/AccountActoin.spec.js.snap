exports[`AccountActions should render the page 1`] = `
<div
  className="account-actions">
  <button
    className="topUp button"
    disabled=""
    onClick={[Function]}>
    TOP UP $1000 ON ACCOUNT
  </button>
  <button
    className="withdraw button"
    disabled=""
    onClick={[Function]}>
    WITHDRAW $1000 FROM ACCOUNT
  </button>
</div>
`;
